telematic_system/historical_data_processing/rosbag2_processing_service/rosbag2_processing_service/rosbag_processor.py

```diff
@@ -0,0 +1,131 @@
+        self.is_processing = False
+
+
+    async def process_rosbag(self,rosbag2_name):
+        measurement_name = rosbag2_name.split('.mcap')[0] # Measurement name is rosbag name without mcap extension
+
+        rosbag_path = os.path.join(self.rosbag_dir, rosbag2_name)
+
+        # Load the rosbag from the config directory
+        for msg in read_ros2_messages(rosbag_path):
+            try:
+                if msg.channel.topic not in self.topic_exclusion_list:
+                    topic = msg.channel.topic
+                    ros_msg = msg.ros_msg
+                    msg_attributes = self.extract_attributes(ros_msg)
+                    msg_timestamp = msg.publish_time_ns
+
+                    record = f"{measurement_name},topic_name={topic},"
+
+                    for attr_name, attr_value in msg_attributes:
+                        if attr_name in self.ignore_fields:
+                            continue
+
+                        elif attr_name in self.to_str_fields:
+                            attr_value = f'"{attr_value}"'
+                            record += f"{attr_name}={attr_value},"
+
+                        elif isinstance(attr_value, list):  # Handle arrays
+                            record += f'{attr_name}="{str(attr_value)}",'
+                        else:
+                            if isinstance(attr_value, str):
+                                attr_value = f'"{attr_value}"'  # Correctly format string values
+                            record += f"{attr_name}={attr_value},"
+
+
+                    # Remove last comma
+                    record = record[:-1]
+                    # Add timestamp at the end
+                    record += f" timestamp={msg_timestamp}"
+                    #Write record to influx
+                    self.write_api.write(bucket=self.influx_bucket, org=self.influx_org, record=record)
+
+            except Exception as e:
+                self.logger.warn(f"Failed to process ros message with exception: " + str(e))
+        self.logger.info(f"Completed rosbag processing for {rosbag2_name}")
+        self.is_processing = False
+
+
+    def extract_attributes(self, obj, parent_attr=None):
+        attributes = []
+        for attr_name in dir(obj):
+            try:
+                if not callable(getattr(obj, attr_name)) and not attr_name.startswith("__") and not attr_name.startswith("_"):
+                    attr_value = getattr(obj, attr_name)
+                    if parent_attr:
+                        attr_name = f"{parent_attr}.{attr_name}"
+                    if hasattr(attr_value, '__dict__'):
+                        # Recursively extract attributes for nested objects
+                        nested_attributes = self.extract_attributes(attr_value, attr_name)
+                        attributes.extend(nested_attributes)
+                    else:
+                        attributes.append((attr_name, attr_value))
+            except Exception as e:
```

Comment:
Same comment about generic `Exception` class.

Reply:
That's fine. I suggest we put that justification as a comment for others that might read the code.